### PR TITLE
Match output types for numeric_range(stop)

### PR DESCRIPTION
**Pull request by bbayles**
Re: Issue #136, this PR makes the one-argument version of `numeric_range()` output items that match the type of the input.

### CODE REVIEW

**Comment by MSeifert04** [comment]
I'm not sure if that follows the "least astonishment" principle. Mainly because the returned types depend on the `stop` argument if and only if only the `stop` argument is given:

```
>>> list(numeric_range(3.59)
[0.0, 1.0, 2.0, 3.0]
>>> list(numeric_range(0, 3.59)
[0, 1, 2, 3]
```

Not a problem per-se but it should be mentioned in the docs. :)

**Comment by bbayles** [comment]
Ah, yes. I put a note in about the type matching the first argument passed in.

File: more_itertools/more.py [hunk 1]
```diff
@@ -1248,19 +1245,21 @@ def numeric_range(*args):
         >>> list(numeric_range(start, stop, step))
         [Fraction(1, 2), Fraction(1, 1), Fraction(3, 2), Fraction(2, 1)]
 
-    Negative steps are supported:
+    If *step* is zero,``ValueError`` is rasied. Negative steps are supported:
 
         >>> list(numeric_range(3, -1, -1.0))
         [3.0, 2.0, 1.0, 0.0]
 
+    The yielded items will have the same type as the first argument.
```
Comment (MSeifert04) [inline more_itertools/more.py]: except when the addition of `start` and a multiple of `step` "coerces" to another type. :)

Comment (bbayles) [inline more_itertools/more.py]: Good counterexample!

**Comment by bbayles** [comment]
We're getting there! Thanks for the attention to detail.

File: more_itertools/more.py [hunk 2]
```diff
@@ -1222,24 +1222,23 @@ def numeric_range(*args):
     """An extension of the built-in ``range()`` function whose arguments can
     be any orderable numeric type.
 
-    If the *start* argument is omitted, it defaults to ``0``. If the *step*
-    argument is omitted, it defaults to ``1``. If *step* is zero,
-    ``ValueError`` is rasied.
-
-    With only *stop* specified:
+    With only *stop* is specified, *start* defaults to ``0`` and *step*
```
Comment (MSeifert04) [inline more_itertools/more.py]: `If` instead of `With`

File: more_itertools/more.py [hunk 3]
```diff
@@ -1248,7 +1247,7 @@ def numeric_range(*args):
         >>> list(numeric_range(start, stop, step))
         [Fraction(1, 2), Fraction(1, 1), Fraction(3, 2), Fraction(2, 1)]
 
-    Negative steps are supported:
+    If *step* is zero, ``ValueError`` is rasied. Negative steps are supported:
```
Comment (MSeifert04) [inline more_itertools/more.py]: ra**is**ed.

Comment (MSeifert04) [inline more_itertools/more.py]: And the second sentence could start with `But negative ...` to emphasis the relation to zero-steps.

File: more_itertools/more.py (hunk 2 above)
Comment (bbayles) [inline more_itertools/more.py]: Eh.

File: more_itertools/more.py (hunk 3 above)
Comment (bbayles) [inline more_itertools/more.py]: Good catch on `raise`d.

File: more_itertools/more.py (hunk 2 above)
Comment (MSeifert04) [inline more_itertools/more.py]: Or `When`?

I'm not a native speaker but `With only stop is specified ...` sounds wrong.

Comment (bbayles) [inline more_itertools/more.py]: The _is_ is unnecessary. This PR is teaching me I should just listen to you the first time.

**Pull request merged**